src/header/map.rs

```diff
@@ -1720,6 +1724,45 @@ impl<T> FromIterator<(HeaderName, T)> for HeaderMap<T>
     }
 }
 
+impl<T> Sealed for HeaderMap<T> {}
```

Comment:
There should probably be an identity impl, `impl<T> HttpTryFrom<HeaderMap<T>> for HeaderMap<T>`. Would that conflict with the generic one below?

Reply:
Yes, `HeaderMap<T>` is itself `IntoIterator<Item=(HttpTryFrom<HeaderName>, HttpTryFrom<HeaderValue>)>`, so `HeaderMap::try_from(header_map)` works as expected.

Reply:
Usually one would expect `TryFrom<Me> for Me` to be basically free, but because that conflicts, it would be surprising that it essentially clones a new `HeaderMap`....

Reply:
This is indeed sub-optimal, and can fixed later when rust adds support for [specialization](https://github.com/rust-lang/rust/issues/37653).
But until then, I think the usability benefits far outweigh this downside. The standard library itself has the same issue in several places, where an implementation is slower than it could be in order to be more generic.